integ-test/src/test/java/org/opensearch/sql/datasource/DataSourceAPIsIT.java

```diff
@@ -317,21 +312,91 @@ public void datasourceLimitTest() throws InterruptedException, IOException {
         errorMessage.get("error").getAsJsonObject().get("details").getAsString());
   }
 
+  @SneakyThrows
+  @Test
+  public void patchDataSourceAPITest() {
```

Comment:
test the upgrade case, which index does not has status field?

Reply:
I have tested manually. Its working fine. Let me see if I could add these test cases.

Reply:
added new test case to load old datasource models..